src/sentry/workflow_engine/migration_helpers/alert_rule.py

```diff
@@ -374,6 +375,99 @@ def migrate_alert_rule(
     )
 
 
+def update_migrated_alert_rule(alert_rule: AlertRule, updated_fields: dict[str, Any]) -> (
+    tuple[
+        DetectorState,
+        Detector,
+    ]
+    | None
+):
+    # TODO: maybe pull this into a helper method?
+    try:
+        alert_rule_detector = AlertRuleDetector.objects.get(alert_rule=alert_rule)
+    except AlertRuleDetector.DoesNotExist:
+        logger.exception(
+            "AlertRuleDetector does not exist",
+            extra={"alert_rule_id": AlertRule.id},
+        )
+        return None
+
+    detector: Detector = alert_rule_detector.detector
+
+    try:
+        detector_state = DetectorState.objects.get(detector=detector)
+    except DetectorState.DoesNotExist:
+        logger.exception(
+            "DetectorState does not exist",
+            extra={"alert_rule_id": AlertRule.id, "detector_id": detector.id},
+        )
+        return None
+
+    updated_detector_fields: dict[str, Any] = {}
+    config = detector.config.copy()
+
+    if "name" in updated_fields:
+        updated_detector_fields["name"] = updated_fields["name"]
+    if "description" in updated_fields:
+        updated_detector_fields["description"] = updated_fields["description"]
+    if "user_id" in updated_fields:
+        updated_detector_fields["owner_user_id"] = updated_fields["user_id"]
+    if "team_id" in updated_fields:
+        updated_detector_fields["owner_team_id"] = updated_fields["team_id"]
```

Comment:
nit: this and the config field section below can be written a little less repetitively, maybe something like

```suggestion
    fields_to_detector_fields = {"name": "name", "description": "description", "user_id": "owner_user_id", "team_id", "owner_team_id"}
    for field, detector_field in fields_to_detector_fields.items():
        if updated_field := updated_fields.get(field):
            updated_detector_fields[detector_field] = updated_field
```